thirtyinch/src/main/java/net/grandcentrix/thirtyinch/LoggingInterceptor.java

```diff
@@ -0,0 +1,189 @@
+/*
+ * Copyright (C) 2017 grandcentrix GmbH
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.grandcentrix.thirtyinch;
+
+import net.grandcentrix.thirtyinch.util.AbstractInvocationHandler;
+
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import android.util.Log;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.lang.reflect.Proxy;
+import java.util.List;
+
+import static net.grandcentrix.thirtyinch.util.AnnotationUtil.getInterfaceOfClassExtendingGivenInterface;
+
+/**
+ * Logs all methods calls and parameters to the bound view interface.
+ */
+public class LoggingInterceptor implements BindViewInterceptor {
+
+    private final static class MethodLoggingInvocationHandler<V> extends AbstractInvocationHandler {
+
+        /**
+         * limit each argument instead of the complete string. This should limit the overall
+         * output to a reasonable length while showing all params
+         */
+        public static final int MAX_LEN_OF_PARAM = 240;
+
+        private TiLog.Logger mLogger;
+
+        private final V mView;
+
+        private MethodLoggingInvocationHandler(V view, @NonNull TiLog.Logger logger) {
+            mView = view;
+            mLogger = logger;
+        }
+
+        @Override
+        public String toString() {
+            return "MethodLoggingProxy@" + Integer.toHexString(this.hashCode()) + "-" + mView
+                    .toString();
+        }
+
+        @Override
+        protected Object handleInvocation(final Object proxy, final Method method,
+                final Object[] args)
+                throws Throwable {
+
+            try {
+                mLogger.log(Log.VERBOSE, TAG, toString(method, args));
```

Comment:
yes, the `InvocationHandler` is just a implementation detail. You add a `LoggingInterceptor` and you expect this interceptor to log something.